mlflow/insights/models/base.py

```diff
@@ -0,0 +1,186 @@
+"""
+Base models and mixins for MLflow Insights.
+
+This module contains the foundational components used by the main entity models.
+"""
+
+from __future__ import annotations
+
+from datetime import datetime, timezone
+from typing import TYPE_CHECKING, Any
+
+import yaml
+from pydantic import BaseModel, ConfigDict, Field, field_validator
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+else:
+    try:
+        from typing import Self
+    except ImportError:
+        from typing_extensions import Self
```

Comment:
Can we always use `from typing_extensions import Self`?

Reply:
Yep!